softlayer-cpi.html.md.erb

```diff
@@ -0,0 +1,156 @@
+
+---
+## <a id='resource-pools'></a> Resource Pools / VM Types
+
+Schema for `cloud_properties` section:
+
+* **Domain** [String, required]: Name of the domain. Example: `softlayer.com`.
+* **VmNamePrefix** [String, required]: Prefix of the vm name. Example: `bosh-softlayer`.
+* **EphemeralDiskSize** [Integer, required]: Ephemeral disk size in gigabyte. Example: `100`.
+* **StartCpus** [Integer, required]: Number of CPUs. Example: `4`.
+* **MaxMemory** [Integer, required]: Memory in megabytes. Example: `8192`.
+* **Datacenter** :
+    * **Name** [String, required]: Name of the datacenter. Example: `lon02`.
+* **HourlyBillingFlag** [Boolean, optional]: If the vm is hourly billing. Default is `false`. 
+* **PrimaryNetworkComponent** :
+    * **NetworkVlan** : 
+        * **Id** [Integer, required]: Id of the public VLAN. Example: `524956`.
+* **PrimaryBackendNetworkComponent** :
+    * **NetworkVlan** : 
+        * **Id** [Integer, required]: Id of the private VLAN. Example: `524954`.
+* **NetworkComponents** :
+    * **MaxSpeed** [Integer, optional]: Max speed of the network in Mbps. Default is `100`.
+
+Example:
+
+```yaml
+resource_pools:
+- name: vms
+  network: default
+  stemcell:
+    url: light-bosh-stemcell-3169-softlayer-esxi-ubuntu-trusty-go_agent
+  cloud_properties:
+    Domain: softlayer.com
+    VmNamePrefix: bosh-softlayer
+    EphemeralDiskSize: 100
+    StartCpus: 4
+    MaxMemory: 8192
+    Datacenter:
+       Name: lon02
+    HourlyBillingFlag: true
+    PrimaryNetworkComponent:
+       NetworkVlan:
+          Id: 524956
+    PrimaryBackendNetworkComponent:
+       NetworkVlan:
+          Id: 524954
+    NetworkComponents:
+    - MaxSpeed: 1000
+```
+
+---
+## <a id='disk-pools'></a> Disk Pools / Disk Types
+
+Currently the CPI does not support any cloud properties for disks.
+
+Example of 100GB disk:
+
+```yaml
+disk_pools:
+- name: disks
+  disk_size: 100_000
+```
+
+---
+## <a id='cloud-config'></a> Example Cloud Config
+
+```yaml
+azs:
+- name: lon02
+  cloud_properties:
+    Datacenter: { Name: lon02  }
+    PrimaryNetworkComponent: { NetworkVlan: { Id: 524956  } }
+    PrimaryBackendNetworkComponent: { NetworkVlan: { Id: 524954  } }
+
+vm_types:
+- name: compilation
+  cloud_properties:
+    Bosh_ip: 10.1.2.3
+    StartCpus:  4
+    MaxMemory:  8192
+    EphemeralDiskSize: 100
+    HourlyBillingFlag: true
+    VmNamePrefix: sl-compilation-worker-
+- name: sl-server
+  cloud_properties:
+    Bosh_ip: 10.1.2.3
+    StartCpus:  4
+    MaxMemory:  8192
+    EphemeralDiskSize: 100
+    HourlyBillingFlag: true
+    VmNamePrefix: sl-
+
+vm_extensions:
+- name: lon02-vlan-id
+  cloud_properties:
+    Datacenter: { Name: lon02  }
+    PrimaryNetworkComponent: { NetworkVlan: { Id: 524956  } }
+    PrimaryBackendNetworkComponent: { NetworkVlan: { Id: 524954  } }
+
+disk_types:
+- name: small
+  disk_size: 50_000
+- name: large
+  disk_size: 500_000
+
+networks:
+- name: default
+  type: dynamic
+  subnets:
+  - az: lon02
```

Comment:
In Softlayer, the concept of availability zone is data center. Here lon02 is a data center name.